crates/sui-core/src/authority_server.rs

```diff
@@ -42,6 +42,7 @@ mod server_tests;
 
 const MIN_BATCH_SIZE: u64 = 1000;
 const MAX_DELAY_MILLIS: u64 = 5_000; // 5 sec
+const MAX_INFLIGHT_CONSENSUS_TRANSACTIONS: u64 = 1000;
```

Comment:
From what I see in the code there is another path that also generates transactions that end up in consensus (the `CheckpointConsensusAdapter`) . Is the assumption that the `MAX_INFLIGHT_CONSENSUS_TRANSACTIONS` cap will only be applied for the externally submitted transactions? (hence eventually handled via the `ConsensusAdapter` ?) - so in practice in consensus there might be more inflight transactions initiated by "us"

Reply:
I think "system" transactions should not be capped to simplify reasoning. But let me see if I should make similar simplifications to `CheckpointConsensusAdapter`.

Reply:
Switched to reusing `ConsensusListerner` for backpressure instead.